Include/internal/pycore_instruments.h

```diff
@@ -0,0 +1,56 @@
+#ifndef Py_INTERNAL_INSTRUMENTS_H
+#define Py_INTERNAL_INSTRUMENTS_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+#ifdef Py_TIMERS
+
+typedef struct timer_records {
```

Comment:
It might help to document each timer, maybe just mention which function is related. Like:

```
     // pymain_main()
     uint64_t main;
```